care/users/api/viewsets/users.py

```diff
@@ -288,8 +288,11 @@ def clear_home_facility(self, request, *args, **kwargs):
             raise ValidationError({"home_facility": "No Home Facility Present"})
         if (
             requesting_user.user_type < User.TYPE_VALUE_MAP["DistrictAdmin"]
-            or requesting_user.user_type in User.READ_ONLY_TYPES
-        ):
+            and (
+                requesting_user.id != user.id
+                and requesting_user.user_type == User.TYPE_VALUE_MAP["Nurse"]
```

Comment:
the conditions are not correct, this would allow any user to clear any users home facility if the requesting user role is higher and belongs to the same district